dashboard-commons/src/main/java/objective/taskboard/timeline/DashboardTimeline.java

```diff
@@ -13,8 +14,10 @@ public DashboardTimeline(Optional<LocalDate> startDate, Optional<LocalDate> endD
         this.endDate = endDate;
     }
 
-    public boolean hasBothDates() {
-        return startDate.isPresent() && endDate.isPresent();
+    public <T> Optional<T> hasBothDates(BiFunction<LocalDate, LocalDate, T> transformer) {
+        return startDate.flatMap(start ->
+                endDate.map(end -> transformer.apply(start, end))
+        );
```

Comment:
It's simpler to understand.
```suggestion
    public boolean hasBothDates() {
        return startDate.isPresent() && endDate.isPresent();
    }
```